Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py

```diff
@@ -32,76 +32,86 @@
-
-    # this event fires whenever we have changes to our universe
+        self.averages = {}
+        
+    def CoarseSelectionFunction(self,coarse):
+        
+        sorted_by_volume = sorted([cf for cf in coarse if cf.Price > 10],
+            key = lambda cf: cf.DollarVolume, reverse=True)
+            
+        selected = {cf.Symbol: cf for cf in sorted_by_volume[:100]}
+        symbols = list(selected.keys())
+        
+        new_symbols = [s for s in symbols if s not in self.averages]
+        if new_symbols:
+            
+            # Get history for all new symbols. If empty, log and return the selected -- WOW!
+            history = self.History(new_symbols, 200, Resolution.Daily)
+            if history.empty:
+                self.Debug(f'Empty history on {self.Time} for {new_symbols}')
+                return Universe.Unchanged   # Continue with previous universe
+                
+            # Only need the closing prices    
+            history = history.close.unstack(0)
+            
+            # Add new item to self.averages
+            for symbol in new_symbols:
+                if symbol in history:
+                    self.averages[symbol] = SelectionData(history[symbol].dropna())
+                    
+        self.symbols = []
+        
+        for symbol, cf in selected.items():
+            symbolData = self.averages[symbol]
+            
+            if symbol not in new_symbols:
+                symbolData.Update(cf.EndTime, cf.AdjustedPrice) 
+                
+            if symbolData.IsReady and symbolData.fast > symbolData.slow:
+                self.symbols.append(symbol)
+                
+        return self.symbols
+        
+        
     def OnSecuritiesChanged(self, changes):
-        # liquidate removed securities
         for security in changes.RemovedSecurities:
-            if security.Invested:
-                self.Liquidate(security.Symbol)
-
-        # we want 20% allocation in each security in our universe
-        for security in changes.AddedSecurities:
-            self.SetHoldings(security.Symbol, 0.1)
-
-
-class SymbolData(object):
-    def __init__(self, symbol):
-        self.symbol = symbol
-        self.tolerance = 1.01
-        self.fast = ExponentialMovingAverage(100)
-        self.slow = ExponentialMovingAverage(300)
-        self.is_uptrend = False
-        self.scale = 0
-
-    def update(self, time, value):
-        if self.fast.Update(time, value) and self.slow.Update(time, value):
-            fast = self.fast.Current.Value
-            slow = self.slow.Current.Value
-            self.is_uptrend = fast > slow * self.tolerance
-
-        if self.is_uptrend:
-            self.scale = (fast - slow) / ((fast + slow) / 2.0)
\ No newline at end of file
+            self.Liquidate(security.Symbol, f'Removed from Universe')
+            
+        self.can_trade = True
+        
+        
+    def OnData(self, data):
+        if self.can_trade:
```

Comment:
Instead of using `self.can_trade` we might want to use `if self.symbols` and clear the list after the trades are placed.